main.go

```diff
@@ -61,6 +63,18 @@ func main() {
 		err = datasetsize.DatasetSize(args)
 	case "list":
 		err = list.List(args)
+	case "login":
+		deviceLogin, err := login.NewDeviceLogin(args)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to contact authentication service, reason: %v\n", err)
+		}
+		err = deviceLogin.Login()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Login failed: %v\n", err)
+		} else {
+			fmt.Println("Login Successful")
+			fmt.Printf("Logged in as %v\n", deviceLogin.UserInfo.Name)
+		}
```

Comment:
Suggestion:
Move all the if-else stuff into the login.go so that it only returns an error or nil like the others.

Reply:
refactored the whole thing, so everything moved in login